packages/server/src/adapters/lambda-utils.ts

```diff
@@ -0,0 +1,30 @@
+import type { APIGatewayProxyEvent } from 'aws-lambda';
+import type { ResponseMetaFn } from '../http/internals/types';
+import type { AnyRouter, inferRouterContext } from '../router';
+
+export type LambdaContext = Record<string, unknown>;
+export type CreateLambdaContextOptions = {
+  event: APIGatewayProxyEvent;
+  context: LambdaContext;
+};
+export type LambdaCreateContextFn<TRouter extends AnyRouter> = ({
+  event,
+  context,
+}: CreateLambdaContextOptions) =>
+  | inferRouterContext<TRouter>
+  | Promise<inferRouterContext<TRouter>>;
+
+export type AWSLambdaOptions<TRouter extends AnyRouter> = {
+  router: TRouter;
+  batching?: {
+    enabled: boolean;
+  };
+  // TODO: Hack 1.
+  onError?: (options: Record<string, unknown>) => void;
+  responseMeta?: ResponseMetaFn<TRouter>;
+} & {
+  /**
+   * @link https://trpc.io/docs/context
+   **/
+  createContext: LambdaCreateContextFn<TRouter>;
+};
```

Comment:
```suggestion
export type LambdaCreateContextOption<TRouter extends AnyRouter> =
  unknown extends inferRouterContext<TRouter>
    ? {
        /**
         * @link https://trpc.io/docs/context
         **/
        createContext?: LambdaCreateContextFn<TRouter>;
      }
    : {
        /**
         * @link https://trpc.io/docs/context
         **/
        createContext: LambdaCreateContextFn<TRouter>;
      };

export interface AWSLambdaOptions<TRouter extends AnyRouter>
  extends HTTPBaseHandlerOptions<TRouter, APIGatewayProxyEvent> & LambdaCreateContextOption<TRouter>
}
```
refer to https://github.com/trpc/trpc/blob/next/packages/server/src/adapters/fetch/types.ts